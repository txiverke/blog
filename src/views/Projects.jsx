// @flow

import React from 'react'
import Helmet from 'react-helmet'

const Projects = () =>
  <div className="app-view">
    <Helmet 
      title="Projects" 
      meta={[
        { name:"description", content: "Personal projects in which I am working currently." },
        { property: "og:title", content: "Personal projects of Xavi VilÃ " },
      ]}
    />
    <h1>Projects</h1>
  </div>

export default Projects