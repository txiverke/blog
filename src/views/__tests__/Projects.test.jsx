// @flow

import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import { Unwrapped as UnwrappedProjects} from '../Projects'

const properties = {
  projects: {
    completed: true,
    data: [],
    message: 'fake project message',
    error: false
  },
  dispatch: jest.fn()
}

function setup() {
  const props = properties

  const enzymeWrapper = mount(
    <MemoryRouter initialEntries={['/test']}>
      <UnwrappedProjects {...props} />
    </MemoryRouter>
  )

  return {
    props,
    enzymeWrapper
  }
}

describe('<Projects />', () => {
  it('should render itself', () => {
    const component = shallow(<UnwrappedProjects {...properties} />)
    expect(component).toMatchSnapshot()
  })

  it('should load data calling dispatch', () => {
    const { enzymeWrapper, props } = setup()  
    const currentCalls = props.dispatch.mock.calls.length
    expect(currentCalls).toBeGreaterThan(0)
  })

  it('should render Loading if no data', () => {
    const component = shallow(<UnwrappedProjects {...properties} />)
    expect(component.find('Loader').length).toEqual(1)
  })

  it('should render component if there is data', () => {
    properties.projects.data = [
      {
        created: "2018-03-14T15:35:36.142Z",
        _id: "5aa",
        title: "react-messages",
        subtitle: "A component to create messages.",
        summary: "'react-messages' is a simple react component which allows you to create messages and configre their look and feel.",
        content: "Here you can test the component and the different options that you can configure.",
        link: "https://www.npmjs.com/package/react-messages",
        background: "89e0fe2351b94644a1edb0e28adb76031521041736137.jpeg",
        __v: 0,
        extra: "React-messages-example",
      }
    ]
    const { enzymeWrapper } = setup()  
    expect(enzymeWrapper.find('Redirect').length).toEqual(1)
  })

})