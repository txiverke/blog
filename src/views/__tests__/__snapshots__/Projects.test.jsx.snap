// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Projects
    dispatch={[Function]}
    projects={
        Object {
            "completed": true,
            "data": Array [],
            "error": false,
            "message": "fake project message",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Loader
        msg="fake project message"
/>,
      "className": "app-view app-view-centered",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "msg": "fake project message",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Loader
          msg="fake project message"
/>,
        "className": "app-view app-view-centered",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "msg": "fake project message",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
